freqtrade/misc.py

```diff
@@ -301,3 +303,44 @@ def remove_entry_exit_signals(dataframe: pd.DataFrame):
     dataframe[SignalTagType.EXIT_TAG.value] = None
 
     return dataframe
+
+
+def sync_to_async_iter(iter):
+    """
+    Wrap blocking iterator into an asynchronous by
+    offloading computation to thread and using
+    pubsub pattern for yielding results
+
+    :param iter: A synchronous iterator
+    :returns: An asynchronous iterator
+    """
+
+    loop = asyncio.get_event_loop()
+    q = asyncio.Queue(1)
+    exception = None
+    _END = object()
+
+    async def yield_queue_items():
+        while True:
+            next_item = await q.get()
+            if next_item is _END:
+                break
+            yield next_item
+        if exception is not None:
+            # The iterator has raised, propagate the exception
+            raise exception
+
+    def iter_to_queue():
+        nonlocal exception
+        try:
+            for item in iter:
+                # This runs outside the event loop thread, so we
+                # must use thread-safe API to talk to the queue.
+                asyncio.run_coroutine_threadsafe(q.put(item), loop).result()
+        except Exception as e:
+            exception = e
+        finally:
+            asyncio.run_coroutine_threadsafe(q.put(_END), loop).result()
+
+    threading.Thread(target=iter_to_queue).start()
```

Comment:
That's not a good solution if i'm honest, i'd prefer we can remove this - at least for now.

This has the potential to spawn of 100ds of threads without control (or tracking) - in addition to having to spawn one thread per consumer.

So far, it's unclear to me what happens to this thread if the consumer disconnects while it's still sending initial messages.
i assume the thread will stall and remain running (though not doing anything) forever - but it'll need full testing.

That PR was 95% done - i'd appreciate if we could not extend the scope of PR's without reason.

This can surely be looked at as a separate PR - but in this PR - please remove.

Reply:
I've reverted this for now for the above reasons.